src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -159,112 +152,242 @@ void CPlayer::kill(HexagonGame& mHexagonGame)
+        if(std::abs(ssvs::getDotProduct(ssvs::getNormalized(pushVel),
+            ssvs::getNormalized(lastPos))) > pushDotThreshold)
+        {
+            pushVel = {0.f, 0.f};
+        }
+    };
 
-    while(wall.isOverlapping(pos))
+    // A setting to allow easier sliding against rows of thin walls, among other possibilities.
+    if(wall.getForgiving())
     {
-        angle += pushAngle;
-        pos = ssvs::getOrbitRad(startPos, angle, radius);
-
-        if(++attempt >= maxAttempts)
+        while(i < 4u)
         {
-            pos = lastPos;
-            angle = lastAngle;
+            collisionPolygon =
+                {&wVertexes[i], &wOldVertexes[i], &wOldVertexes[j], &wVertexes[j]};
 
-            return true;
+            if(pointInPolygonPointers(collisionPolygon, lastPos.x, lastPos.y))
+            {
+                getPushVel();
+                break;
+            }
+
+            j = i++;
+        }
+    }
+    else
+    {
+        while(i < 4u)
+        {
+            collisionPolygon =
+                {&wVertexes[i], &wOldVertexes[i], &wOldVertexes[j], &wVertexes[j]};
+
+            // Check for a killing side, this takes priority.
+            if(pointInPolygonPointers(collisionPolygon, pos.x, pos.y))
+            {
+                // Average velocity of the vertexes, we cannot use
+                // getLineCircleClosestIntersection() cause this side
+                // might not have intersected the player circle in the previous frame.
+                pushVel = ((wVertexes[i] - wOldVertexes[i]) +
+                    (wVertexes[j] - wOldVertexes[j])) / 2.f;
+
+                // If wall is coming in too steep kill now.
+                if(std::abs(ssvs::getDotProduct(ssvs::getNormalized(pushVel),
+                    ssvs::getNormalized(pos))) > killDotThreshold)
+                {
+                    return true;
+                }
+            }
+
+            // Check for a pushing side.
+            if(pointInPolygonPointers(collisionPolygon, lastPos.x, lastPos.y))
+            {
+                getPushVel();
+                break;
+            }
+
+            j = i++;
         }
     }
 
+    // Player is not moving exit now.
+    if(!mHexagonGame.getInputMovement() && !forcedMove)
+    {
+        pos += pushVel;
+        pos += ssvs::getNormalized(pos - prePushPos) *
+            (2.f * collisionPadding);
+        angle = ssvs::getRad(pos);
+        updatePosition(mHexagonGame, mFT);
+        return wall.isOverlapping(pos);
+    }
+
+    // If alive try to find a close enough safe position.
+    sf::Vector2f testPos{lastPos + pushVel};
+    if(wall.isOverlapping(testPos) ||
+       !checkWallCollisionEscape(wall, testPos, mRadiusSquared))
+    {
+        return true;
+    }
+
+    // If player survived apply test position and add a little padding.
+    pos = testPos + ssvs::getNormalized(testPos - prePushPos) * collisionPadding;
+    angle = ssvs::getRad(pos);
+    updatePosition(mHexagonGame, mFT);
```

Comment:
Ditto, this is also pretty arcane to me (the maths), but I'm gonna trust (and test)